### PR TITLE
feature: support ngsi-ld context broker #82

### PR DESCRIPTION
**Pull request by vraybaud**

### CODE REVIEW

File: test/tools/ngsiUtils.js [hunk 1]
```diff
@@ -127,17 +144,25 @@ function discoverContextAvailability(host, port, service, subservice, id, type,
...
-    return {
-        query: queryEntity.bind(null, host, port, service, subservice),
-        update: updateEntity.bind(null, host, port, service, subservice),
-        discover: discoverContextAvailability.bind(null, host, port, service, subservice)
-    };
+    if (ngsiVersion === 'ld')
+        return {
+            query: queryEntityNgsiLD.bind(null, host, port, service, subservice),
+            update: updateEntity.bind(null, host, port, service, subservice),
+            discover: discoverContextAvailability.bind(null, host, port, service, subservice)
```
Comment (bobeal) [inline test/tools/ngsiUtils.js]: Those two queries won't work with a NGSI-LD context broker. 

They are not used by the agent?

File: test/unit/plugin-config-test.js [hunk 2]
```diff
@@ -84,17 +85,20 @@ describe('Plugin configuration test', function() {
...
-                    ngsiClient.query('sigApp3', 'SIGFOX', [], function(error, response, body) {
+                    ngsiClient.query('sigApp3', 'Device', [], function(error, response, body) {
                         var attributes;
 
                         should.not.exist(error);
                         should.exist(body);
                         should.not.exist(body.errorCode);
 
-                        attributes = body.contextResponses[0].contextElement.attributes;
+                        console.log(body);
```
Comment (bobeal) [inline test/unit/plugin-config-test.js]: Don't forget to remove that

File: package.json [hunk 3]
```diff
@@ -37,7 +37,7 @@
     "command-node": "0.1.1",
     "request": "2.88.0",
     "async": "1.5.2",
-    "iotagent-node-lib": "git://github.com/telefonicaid/iotagent-node-lib.git#master",
+    "iotagent-node-lib": "git://github.com/telefonicaid/iotagent-node-lib.git#feature/842_ngsi_ld",
```
Comment (bobeal) [inline package.json:40]: We need a strategy for this. Maybe keep the PR open until the NGSI-LD branch is merged in the node lib and then merge this PR?

File: test/tools/ngsiUtils.js (hunk 1 above)
Comment (vraybaud) [reply test/tools/ngsiUtils.js]: They are not used by the agent indeed. They are test utils but only the query one is used (and removed when integrating nock...)

File: test/unit/plugin-config-test.js (hunk 2 above)
Comment (vraybaud) [reply test/unit/plugin-config-test.js]: Indeed :+1:

File: package.json (hunk 3 above)
Comment (vraybaud) [reply package.json:40]: Yes I share the same view as you do. Let's wait for iotagent-node-lib to be merged before trying to merge anything.

File: test/tools/ngsiUtils.js (hunk 1 above)
Comment (bobeal) [reply test/tools/ngsiUtils.js]: So maybe remove them in the "nock branch" then rebase this branch on top of it.

Comment (vraybaud) [reply test/tools/ngsiUtils.js]: I did it the oppositve way, ngsi-ld-nock is based on top of this branch.

**Pull request closed without merging**